mlflow/models/wheeled_model.py

```diff
@@ -0,0 +1,237 @@
+        )
+
+        # Update conda.yaml with wheels
+        self._update_conda_env(pip_deps, conda_env_path)
+
+        # Update MLModel File
+        mlflow_model = self._update_mlflow_model(
+            original_model_metadata=model_metadata, mlflow_model=mlflow_model
+        )
+        mlflow_model.save(model_metadata_path)
+        return mlflow_model
+
+    def _update_conda_env(self, new_pip_deps, conda_env_path):
+        """
+        Updates the list pip packages in the conda.yaml file to the list of wheels in the wheels
+        directory.
+        {
+            "name": "env",
+            "channels": [...],
+            "dependencies": [
+                ...,
+                "pip",
+                {"pip": [...]},  <- Overwrite this with list of wheels
+            ],
+        }
+        :param new_pip_deps: List of pip dependencies as wheels
+        :param conda_env_path: Path to conda.yaml file in the model directory
+        """
+        with open(conda_env_path) as f:
+            conda_env = yaml.safe_load(f)
+
+        new_conda_env = _overwrite_pip_deps(conda_env, new_pip_deps)
+
+        with open(conda_env_path, "w") as out:
+            yaml.safe_dump(new_conda_env, stream=out, default_flow_style=False)
+
+    def _update_mlflow_model(self, original_model_metadata, mlflow_model):
+        """
+        Modifies the MLModel file to reflect updated information such as the run_id,
+        utc_time_created. Additionally, this also adds `wheels` to the MLModel file to indicate that
+        this is a `wheeled` model.
+        :param original_model_file_path: The model metadata stored in the original MLmodel file.
+        :param mlflow_model: :py:mod:`mlflow.models.Model` configuration of the newly created
+                                wheeled model
+        """
+
+        run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
+        if mlflow_model is None:
+            mlflow_model = Model(run_id=run_id)
+
+        original_model_metadata.__dict__.update(
+            {k: v for k, v in mlflow_model.__dict__.items() if v}
+        )
+        mlflow_model.__dict__.update(original_model_metadata.__dict__)
+        mlflow_model.artifact_path = WheeledModel.get_wheel_artifact_path(
+            mlflow_model.artifact_path
+        )
+
+        mlflow_model.wheels = {_PLATFORM: platform.platform()}
+        return mlflow_model
+
+    def _download_wheels(self, pip_requirements_path, dst_path):
+        """
+        Downloads all the wheels of the dependencies specified in the requirements.txt file.
+        :param pip_requirements_path: Path to requirements.txt in the model directory
+        :param dst_path: Path to the directory where the wheels are to be downloaded
+        """
+        if not os.path.exists(dst_path):
+            os.makedirs(dst_path)
+
+        download_command = (
+            f"python -m pip wheel --only-binary=:all: --wheel-dir={dst_path} -r"
```

Comment:
```suggestion
            f"{sys.executable} -m pip wheel --only-binary=:all: --wheel-dir={dst_path} -r"
```

It might be safe to use [`sys.executable`](https://docs.python.org/3/library/sys.html#sys.executable) here because `python` might point to python 2.